docs/tp/01.md

```diff
@@ -2,4 +2,486 @@
+
+```rust
+struct User {
+    active: bool,
+    username: String,
+    email: String,
+    sign_in_count: u64,
+}
+```
+Pour utiliser une structure après l'avoir définie, nous créons une **instance** de cette structure en spécifiant des valeurs concrètes pour chacun des champs. Nous créons une instance en indiquant le **nom** de la structure, puis ajoutons des accolades contenant des **paires clé : valeur**, où les clés sont les noms des champs et les valeurs sont les données que nous voulons stocker dans ces champs.
+
+```rust
+fn main() {
+    let user1 = User {
+        active: true,
+        username: String::from("someusername123"),
+        email: String::from("someone@example.com"),
+        sign_in_count: 1,
+    };
+}
+```
+
+Pour acceder a un certain membre du struct on utilise cette syntaxe:
+
+```rust
+fn main() {
+    let mut user1 = User {
+        active: true,
+        username: String::from("someusername123"),
+        email: String::from("someone@example.com"),
+        sign_in_count: 1,
+    };
+ 
+    user1.email = String::from("anotheremail@example.com");
+}
+```
+:::tip
+Notez que l'instance entière doit être **modifiable** ; Rust **ne nous permet** pas de marquer uniquement certains champs comme mutables!
+:::
+
+Comme pour toute expression, nous pouvons construire une nouvelle instance de la structure en tant que dernière expression dans le corps de la fonction pour renvoyer implicitement cette nouvelle instance.
+
+```rust
+fn build_user(email: String, username: String) -> User {
+    User {
+        active: true,
+        username: username,
+        email: email,
+        sign_in_count: 1,
+    }
+}
+```
+
+#### Tuple structs
+Rust prend également en charge les structures qui ressemblent aux tuples, appelées **tuple structs**. Les structures de tuple ont la signification supplémentaire fournie par le nom de la structure mais n'ont pas de noms associés à leurs champs ; au lieu de cela, ils ont juste les types des champs. Les structures de tuple sont utiles lorsque vous souhaitez donner un nom à l'ensemble du tuple et lui donner un type différent des autres tuples, et lorsque nommer chaque champ comme dans une structure régulière serait verbeux ou redondant.
+
+```rust
+struct Color(i32, i32, i32);
+struct Point(i32, i32, i32);
+ 
+fn main() {
+    let black = Color(0, 0, 0);
+    let origin = Point(0, 0, 0);
+}
+```
+:::tip 
+Pour comprendre mieux, lire le **chapitre 5** qui se trouve dans la **documentation** au début du TP!
+:::
+
+### Enums
+Les énumérations, également appelées `enum`, vous permettent de définir un type en énumérant ses variantes possibles.
+Comment définir une énumération :
+
+```rust
+enum IpAddrKind {
+    V4,
+    V6,
+}
+```
+
```

Comment:
Add details about `match` similar to https://embedded-rust-101.wyliodrin.com/docs/lab/00#match.